### PR TITLE
[Live-Site] Add 100ms sdk to join room & Screen share

### LINKED ISSUE #491: [Live-Site] Add the Join functionality in live share
### AC
- Guest and Host  users will be able to join the live share just by putting their name
- The role of the user will be determined from the query param (role)
- Integrate 100ms SDK to join the dummy room which will be replaced with backed API in future

### UX
https://www.figma.com/file/2VtPGJH61Qp0iarkKW6KTO/Live-Site-Detailed-Designs?type=design&node-id=0-1&t=aWR2yCHfQdp1BaF4-0

### PR DESCRIPTION
**Pull request by SanketDhabarde**
### What is the change?
We integrated 100ms to join the live session and share the screen by the host. The API which is used to get the token or join the room will be replaced with the backed APIs when they are ready.

### Is Development Tested?

- [x] Yes
- [ ] No

#### Join
https://github.com/Real-Dev-Squad/website-www/assets/40385118/c40f3d7a-6711-402e-b843-fb0d9b91c955
#### Screenshare
https://github.com/Real-Dev-Squad/website-www/assets/40385118/710c87ec-573e-4f5b-8280-434132662e52

### CODE REVIEW

**Review by satyam73 — CHANGES_REQUESTED** [review]

File: app/controllers/live.js [hunk 1]
```diff
@@ -27,8 +35,10 @@ export default class LiveController extends Controller {
...
   @action clickHandler(e) {
     e.preventDefault();
-    //TODO: Add funtionality to join live session
-    if (this.name) {
+    const isGuest = this.role === this.ROLES.guest;
+    const isHost = this.role === this.ROLES.host;
+    if (this.name && (isGuest || isHost)) {
+      this.liveService.joinSession(this.name, this.role);
       this.isJoined = true;
       this.name = '';
```
[the remark this answers is not in the document]
Comment (SanketDhabarde) [reply app/controllers/live.js:43]: That's generally what we do right when we set up any form. So if the user having some error they don't have to clear the name manually and type again.

File: app/controllers/live.js [hunk 2]
```diff
@@ -13,6 +17,10 @@ export default class LiveController extends Controller {
   @tracked isLoading = true;
   @tracked name = '';
   @tracked isJoined = false;
+  @tracked role = null;
```
[the remark this answers is not in the document]
Comment (SanketDhabarde) [reply app/controllers/live.js:20]: This role will be taken from queryParams. So if we don't add the queryParams it's basically null. From that idea, I have kept it null. An empty string is still some value.

File: app/controllers/live.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (satyam73) [reply app/controllers/live.js:43]: got it

**Review by satyam73 — APPROVED** [review]
LGTM

File: app/controllers/live.js [hunk 3]
```diff
@@ -18,6 +19,7 @@ export default class LiveController extends Controller {
   @tracked name = '';
   @tracked isJoined = false;
   @tracked role = null;
+  @globalRef('videoEl') videoEl;
```
Comment (satyam73) [inline app/controllers/live.js:22]: NIT : remove this unused declaration

**Review by satyam73 — APPROVED** [review]
LGTM👍✨
Good work @SanketDhabarde

**Review by rohan09-raj — APPROVED** [review]
Great work @SanketDhabarde 💯 💯 Merging your PR

**Pull request merged**